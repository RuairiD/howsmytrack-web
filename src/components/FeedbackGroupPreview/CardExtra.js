import React from "react";
import dateFormat from "dateformat";

import { Button, Typography } from "antd";
import { Div } from "lemon-reset";

const formatTimeCreated = (timeCreated) => dateFormat(
    new Date(Date.parse(timeCreated)),
    "mmmm dS yyyy",
);

const CardExtra = ({ timeCreated }) => (
    <Div style={{ display: "flex", marginTop: "1em" }}>
        <Typography.Text style={{
            marginRight: "auto",
            marginTop: "auto",
            marginBottom: "auto",
        }}
        >
            {formatTimeCreated(timeCreated)}
        </Typography.Text>
        <Typography.Text style={{
            marginLeft: "auto",
            marginTop: "auto",
            marginBottom: "auto",
        }}
        >
            <Button type="link">View Group</Button>
        </Typography.Text>
    </Div>
);

export default CardExtra;
