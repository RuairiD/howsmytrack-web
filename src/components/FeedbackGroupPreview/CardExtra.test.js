import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import CardExtra from "./CardExtra";

describe("CardExtra", () => {
    it("renders the footer for a FeedbackGroupPreview with the formatted date", () => {
        const wrapper = shallow(
            <CardExtra timeCreated={123456} />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
