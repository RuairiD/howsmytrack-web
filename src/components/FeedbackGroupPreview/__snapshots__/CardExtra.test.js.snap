// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardExtra renders the footer for a FeedbackGroupPreview with the formatted date 1`] = `
<Div
  className=""
  style={
    Object {
      "display": "flex",
      "marginTop": "1em",
    }
  }
>
  <Text
    style={
      Object {
        "marginBottom": "auto",
        "marginRight": "auto",
        "marginTop": "auto",
      }
    }
  >
    January 1st 123456
  </Text>
  <Text
    style={
      Object {
        "marginBottom": "auto",
        "marginLeft": "auto",
        "marginTop": "auto",
      }
    }
  >
    <Button
      block={false}
      ghost={false}
      htmlType="button"
      loading={false}
      type="link"
    >
      View Group
    </Button>
  </Text>
</Div>
`;
